test/internals/initialConfig.test.js

```diff
@@ -198,8 +198,8 @@ test('Original options must not be altered (test deep cloning)', t => {
   t.strictEqual(Object.isFrozen(fastify.initialConfig), true)
 
   // originalOptions must not have been altered
-  t.deepEqual(originalOptions.https.key, originalOptionsClone.https.key)
-  t.deepEqual(originalOptions.https.cert, originalOptionsClone.https.cert)
+  t.deepEqual(originalOptions.https.key, Object.values(originalOptionsClone.https.key))
+  t.deepEqual(originalOptions.https.cert, Object.values(originalOptionsClone.https.cert))
```

Comment:
I would prefer the base64 approach. I’m not conviced this is actually correct.

Reply:
I tried with the base64 approach, but wasn't working (in my implementation), because of the different structure in cloned objects; I can commit my previous test, just to understand a better fix.
Note that I hadn't to force the '--no-esm' flag in Tap.

Reply:
> I tried with the base64 approach, but wasn't working (in my implementation), because of the different structure in cloned objects; I can commit my previous test, just to understand a better fix.

Then we have a bug in the test or in the cloning. What's the difference?

Reply:
found the problem: originalOptions.https.key and cert are Buffer, but its clone objects are normal objects.I see that our deepClone implementation delegates the clone to [rfdc](https://www.npmjs.com/package/rfdc), in the meantime I can try to ask them for current behavior.
Sorry a question, in failing test, we get `Buffer` instances from `readFileSync`, but if instead we add an encoding there, would be good ?

Reply:
Ahum, I think this is a bug on our logic. @delvedor what do you think?

Reply:
> in failing test, we get Buffer instances from readFileSync, but if instead we add an encoding there to instead get an encoded string (to trasform later to base64 if needed), would be good ?

I think that this might solve the issue.

Reply:
changes done, now all is good, right ?